### DISCUSSION TITLE
Prevent highlighting all edits?

### DISCUSSION THREAD

**Discussion by ofgrenudo** [helix-editor/helix#8005] [Q&A]
### Summary

Hello All,

Ive been using helix and have absolutely loved it so far. Although im running into an issue where it is highlighting all of my edits and kind of gets annoying after a while. Is there a way to prevent this behaviour?

![image](https://github.com/helix-editor/helix/assets/117940901/38b84c7d-0766-4135-8750-f6419f66e67c)

### Reproduction Steps

 

Type into file

I expected this to happen:

The text to appear normally

Instead, this happened:

The text appears, but highlighted.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
2023-08-19T10:59:28.499 helix_view::editor [ERROR] Failed to initialize the LSP for `source.yaml` { cannot find binary path }
2023-08-19T11:02:59.269 helix_view::editor [ERROR] editor error: no such command: 'vplsi'
2023-08-19T11:03:09.332 helix_view::editor [ERROR] Failed to initialize the LSP for `source.toml` { cannot find binary path }
```

</details>

### Platform

Windows

### Terminal Emulator

Window Term with ZSH and MSYS

### Helix Version

helix 23.05 (7f5940be)

**Comment by the-mikedavis** [comment]
That highlighting is the selection. Inserting with `i` preserves selections and inserts characters before the cursor while appending with `a` extends the selection by any inserted characters, as you're seeing in the screenshot. The selection behavior while inserting is an important part of the select-then-act paradigm and can't be turned off.

**Comment by donaldcallen** [comment]
I have found this a bit annoying as well, but I realized that the choice of theme has major role in this. I was using dark_high_contrast, which produces a very bright white highlighting of the text in input mode. Switching to another theme that uses an acceptable highlighting color solved the problem.
